engine/robot_logic/traversal.py

```diff
@@ -347,3 +347,36 @@ def turn_to_target_heading(
     robot_state.enable_obstacle_avoidance = True
     if not robot_state.is_sim:
         robot_state.motor_controller.spin_motors(0, 0)
+
+    def ackerman_calculations (left_ang, right_ang, inside_ang, wheel_base, 
+                               steering_ratio):
+        #calculate ideal wheel angle
+
+        ackerman_percentage = 0 #This value needs to be determined by mechanical
+
+        track_width = wheel_base * ((1/np.tan(left_ang) - (1/np.tan(right_ang))))
+        ackerman_ratio = (inside_ang / steering_ratio)
+        new_left_ang = 1/ np.tan((wheel_base * np.tan(ackerman_ratio)) / (wheel_base + (.5 * track_width )(np.tan(ackerman_ratio))))
+        new_right_ang = 1/ np.tan((wheel_base * np.tan(ackerman_ratio)) / (wheel_base - (.5 * track_width )(np.tan(ackerman_ratio))))
+        new_inside = new_left_ang #(change to new_right_ang based on above logic) ^
+        outside_ang = new_inside - ackerman_percentage * (new_inside - ackerman_ratio)
+
+        return [track_width, ackerman_ratio, new_left_ang, new_right_ang, new_inside, outside_ang]
+
+    def parallel_ackerman_right (left_ang, right_ang, inside_ang, wheel_base, 
+                               steering_ratio):
+        
+        data = ackerman_calculations(left_ang, right_ang, inside_ang, wheel_base, 
+                               steering_ratio)
+        
+            #continue to implement
```

Comment:
Maybe just raise an exception with a comment TODO so you know what parts need to be finished and if you run any test cases with this the exception will tell you you have to finish the function.